internal/config/types/sql.go

```diff
@@ -103,7 +110,7 @@ func (s *SQL) checkSSL() {
 		s.SSL = &ssl{sslBase: sslBase{Mode: sslDisabledMode}}
 		return
 	}
-	if s.SSL.Cert == "" || s.SSL.Key == "" || s.SSL.RootCert == "" {
+	if s.SSL.Cert == "" && s.SSL.Key == "" && s.SSL.RootCert == "" {
```

Comment:
As only the `SSLRootCert` is required i would change the check 
```suggestion
	if s.SSL.RootCert == "" {
```

Reply:
and i would move the call to `checkSSL` to line 81